test/gtest/gpu_mha_backward.cpp

```diff
@@ -0,0 +1,441 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "get_handle.hpp"
+#include "mha_helper.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include "../workspace.hpp"
+#include "../tensor_util.hpp"
+
+#include <miopen/miopen.h>
+#include <miopen/solution.hpp>
+
+#include <gtest/gtest.h>
+
+#include <map>
+#include <memory>
+#include <variant>
+#include <vector>
+
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+using namespace miopen;
+
+namespace {
+inline bool CheckFloatArg(std::string_view arg)
+{
+    const std::string& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
+    return tmp.empty() || tmp == arg;
+}
+
+struct TensorStruct
+{
+    template <typename T>
+    TensorStruct(tensor<T>&& val) : m_cpu_tensor(std::move(val))
+    {
+    }
+
+    TensorStruct(const TensorStruct&) = delete;
+    TensorStruct& operator=(const TensorStruct&) = delete;
+
+    ~TensorStruct() = default;
+
+    std::variant<tensor<float>, tensor<int>> m_cpu_tensor;
+    Allocator::ManageDataPtr m_gpu_buffer;
+};
+
+struct TestCase
+{
+    size_t n;
+    size_t h;
+    size_t s;
+    size_t d;
+    float dropout;
+};
+
+inline std::vector<TestCase> GetSmokeTestCases()
+{
+    if(CheckFloatArg("--float"))
+    {
+        return {
+            {9, 8, 8, 8, 0.0f},
+            {1, 2, 4, 5, 0.0f},
+            {2, 1, 5, 4, 0.0f},
+            {4, 2, 1, 3, 0.0f},
+            {5, 3, 4, 1, 0.0f},
+            {1, 2, 65, 5, 0.0f},
+            {2, 1, 67, 4, 0.0f},
+            {8, 7, 68, 1, 0.0f},
+            {1, 2, 257, 5, 0.0f},
+            {2, 1, 259, 4, 0.0f},
+            {8, 7, 270, 1, 0.0f},
+            {1, 1, 1, 1, 0.0f},
+            {3, 5, 32, 7, 0.8f},
+            {2, 2, 64, 128, 0.8f},
+            {2, 1, 128, 4, 0.8f},
+            {2, 7, 256, 31, 0.8f},
+        };
+    }
+    else
+    {
+        return {};
+    }
+}
+
+inline std::vector<TestCase> GetFullTestCases()
+{
+    if((miopen::IsUnset(ENV(MIOPEN_TEST_ALL)) || miopen::IsEnabled(ENV(MIOPEN_TEST_ALL))) &&
+       CheckFloatArg("--float"))
```

Comment:
```suggestion
    if(miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)))
        return {};

    if(CheckFloatArg("--float"))
```

[Suggestion] This way it would be harder to accidentally add an else that ignores `MIOPEN_TEST_ALL`